tracer.go

```diff
@@ -205,7 +212,11 @@ func (t *Tracer) startSpanWithOptions(
 		newTrace = true
 		ctx.traceID.Low = t.randomID()
 		if t.options.gen128Bit {
-			ctx.traceID.High = t.randomID()
+			if t.options.highTraceIDGenerator != nil {
```

Comment:
the branching is not necessary, you can default highTraceIDGenerator to randomID() in the constructor